drivers/SmartThings/zwave-switch/src/zwave-dual-switch/init.lua

```diff
@@ -45,29 +45,52 @@ local function can_handle_zwave_dual_switch(opts, driver, device, ...)
   return false
 end
 
-local function device_added(self, device)
-  device:refresh()
+local function find_child(parent, src_channel)
+  if src_channel == 1 then
+    return parent
+  else
+    return parent:get_child_by_parent_assigned_key(string.format("%02X", src_channel))
+  end
 end
 
-local function endpoint_to_component(device, endpoint)
-  if endpoint == 2 then
-    return "switch1"
+local function generate_child_name(parent_label)
+  if string.sub(parent_label, -1) == '1' then
+    return string.format("%s2", string.sub(parent_label, 0, -2))
   else
-    return "main"
+    return string.format("%s 2", parent_label)
   end
 end
 
-local function component_to_endpoint(device, component)
-  if component == "switch1" then
-    return {2}
-  else
-    return {1}
+local function device_added(driver, device)
+  if device.network_type ~= st_device.NETWORK_TYPE_CHILD then
+    local name = generate_child_name(device.label)
+    local metadata = {
+      type = "EDGE_CHILD",
+      label = name,
+      profile = "switch-binary",
+      parent_device_id = device.id,
+      parent_assigned_child_key = string.format("%02X", 2),
+      vendor_provided_label = name
+    }
+    driver:try_create_device(metadata)
   end
+  device:refresh()
 end
 
-local function map_components(self, device)
-  device:set_endpoint_to_component_fn(endpoint_to_component)
-  device:set_component_to_endpoint_fn(component_to_endpoint)
+local function endpoint_to_component(device, endpoint)
+  return "main"
+end
```

Comment:
this is the default, so you shouldn't need to set anything in `init` since it won't use the base driver's method